indexing-service/src/test/java/org/apache/druid/indexing/common/task/IndexTaskTest.java

```diff
@@ -1783,6 +1788,36 @@ public void testIndexTaskWithSingleDimPartitionsSpecThrowingException() throws E
     task.isReady(createActionClient(task));
   }
 
+  @Test
+  public void testShardSpecSelectionWithNullPartitionDimension()
```

Comment:
Thank you for adding a unit test! This test doesn't seem testing the `IndexTask` but testing `ShardSpecs`. Would you please move the `ShardSpecs` out of `IndexTask` and add a new class `ShardSpecsTest`? Then we can move this test to `ShardSpecsTest`.

Reply:
Fixed